python/marvin/contrib/vacs/mangagalaxyzoo.py

```diff
@@ -0,0 +1,90 @@
+# !usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Modified from mangahi.py by Karen Masters. 
+
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+import astropy
+import astropy.units as u
+import marvin.tools
+from marvin.tools.quantities.spectrum import Spectrum
+from marvin.utils.general.general import get_drpall_table
+from marvin.utils.plot.scatter import plot as scatplot
+
+from .base import VACMixIn
+
+
+class GZVAC(VACMixIn):
+    """Provides access to the MaNGA Galaxy Zoo Morphology VAC.
+
+    VAC name: GZ
+
+    URL: link
+
+    Description: Returns Galaxy Zoo morphology for MaNGA galaxies
+
+    Authors: Coleman Krawczyk, Karen Masters and the Galaxy Zoo Team.
+
+    """
+
+    # Required parameters
+    name = 'mangagalaxyzoo'
+    description = 'Returns Galaxy Zoo morphology'
+    version = {'MPL-7': 'v1_0_1','MPL-8': 'v1_0_1',
+               'DR15': 'v1_0_1','DR16': 'v1_0_1'}
+
+    # optional Marvin Tools to attach your vac to
+    include = (marvin.tools.cube.Cube,
+               marvin.tools.maps.Maps,
+               marvin.tools.modelcube.ModelCube)
+
+    # Required method
+    def get_data(self, parent_object):
+
+        # get any parameters you need from the parent object
+        plateifu = parent_object.plateifu
+        release = parent_object.release
+
+        # define the variables to build a unique path to your VAC file
+        path_params = {'ver': self.version[release]}
+
+        # get_path returns False if the files do not exist locally
+        gzfile = self.get_path('mangagalaxyzoo', path_params=path_params)
+
+        # download the vac from the SAS if it does not already exist locally
+        if not allfile:
+            gzfile = self.download_vac('mangagalaxyzoo', path_params=path_params)
+
+        return gzdata
+
+
+class GZData(object):
```

Comment:
If all you want to return is the galaxy zoo row data for a given target, you may not need a full class object.  You could probably simplify the content of this class down into a few lines, and move it into the `get_data` method.

In principle, what you want to do is 
- open the `gzfile` FITS file (line 77)
- select and return the proper target row (lines 86, 87)
- format it they way you want to present it to the user, e.g. FITS rec, dictionary
- return that data in `get_data`

Reply:
Yes that's true.

Reply:
(but I don't really know how to do that sorry).

Reply:
OK maybe I did this - take a look. I didn't delete the GZdata class yet.

Reply:
@karenlmasters You were quite close.  I made a few changes to get it working.  I had to switch to mangaid since the VAC doesn't contain a `plateifu` column.  Please have a look and make sure it works for you and returns what you want.  It currently gets the row of data from the FITS and returns it.  So it returns as a FITS record array.